adapters/brave/brave.go

```diff
@@ -0,0 +1,164 @@
+package brave
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb/v16/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint *template.Template
+}
+
+// Builder builds a new instance of the Brave adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	uri, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := &adapter{
+		endpoint: uri,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+	var braveExt *openrtb_ext.ExtImpBrave
+	var err error
+
+	for i, imp := range request.Imp {
+		braveExt, err = a.getImpressionExt(&imp)
```

Comment:
Optimization suggestion: if there are more than one impressions in request this code will be executed for every imp, and `braveExt` will have data from the last imp in request. Do you think it makes sense to move it outside of the for loop and execute `getImpressionExt` just once?

Reply:
> > This code is identical to #2399. If these are aliases of the same underlying bidder, it's better to implement it just once and use an alias. Less code to maintain.
> 
> Hi @thebraveio, just checking in to see if you can shed some light on this. Thanks!

Yes , I developed code for both, but its different companies and strategy, not sure I will support both of them in future

Reply:
> > This code is identical to #2399. If these are aliases of the same underlying bidder, it's better to implement it just once and use an alias. Less code to maintain.
> 
> Hi @thebraveio, just checking in to see if you can shed some light on this. Thanks!

Yes , I developed code for both, but its different companies and strategy, not sure I will support both of them in future